packages/lazy-import/lib/index.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * External dependencies
+ */
+const npa = require( 'npm-package-arg' );
+const semver = require( 'semver' );
+const execa = require( 'execa' );
+const { join } = require( 'path' );
+const { createHash } = require( 'crypto' );
+
+/**
+ * @typedef WPLazyImportOptions
+ *
+ * @property {()=>void} onInstall Callback to invoke when install starts.
+ */
+
+/**
+ * Returns an md5 hash of the given string.
+ *
+ * @param {string} text Text for which to generate hash.
+ *
+ * @return {string} md5 hash of string.
+ */
+function md5( text ) {
+	return createHash( 'md5' )
+		.update( text )
+		.digest( 'hex' );
+}
+
+/**
+ * Given an arg string as would be passed to `npm install`, returns a string to
+ * use as the alias name of the package to install locally if needed.
+ *
+ * @param {string} arg A string that you might pass to `npm install`.
+ *
+ * @return {string} Module to use as local package alias.
+ */
+function getLocalModuleName( arg ) {
+	return '@wordpress/lazy-import.' + md5( arg );
+}
+
+/**
+ * Installs npm package by arg name at the given prefix path. Creates the prefix
+ * path if it doesn't yet exist.
+ *
+ * @param {string} arg   Package install arg (name and version specification).
+ * @param {string} alias Alias to use for local module name.
+ *
+ * @return {Promise<void>} Promise resolving once package is installed.
+ */
+async function install( arg, alias ) {
+	await execa( 'npm', [ 'install', '--no-save', alias + '@npm:' + arg ] );
```

Comment:
Nice trick, I like it 😃 It means it uses native npm logic, so convenient!

It also means that npm 6.9.0+ is required.

Reply:
>It also means that npm 6.9.0+ is required.

Yeah, that's true. I can clarify it in the documentation.

Reply:
> > It also means that npm 6.9.0+ is required.
> 
> Yeah, that's true. I can clarify it in the documentation.

Documented in e67a81f0c3.